engine/common/splitter/network/network.go

```diff
@@ -0,0 +1,96 @@
+package network
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	splitterEngine "github.com/onflow/flow-go/engine/common/splitter"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/network"
+	"github.com/rs/zerolog"
+)
+
+type Network struct {
+	mu        sync.RWMutex
+	net       module.Network
```

Comment:
can't we instead do this?
```suggestion
           module.Network
```
this will splitter/network a type of Network

Reply:
But we also need to actually keep an instance of the `network` since we proxy engine registrations to it